plugin/src/main/java/org/opensearch/ml/model/MLModelCache.java

```diff
@@ -29,16 +30,30 @@ public class MLModelCache {
     private @Setter(AccessLevel.PROTECTED) @Getter(AccessLevel.PROTECTED) MLModelState modelState;
     private @Setter(AccessLevel.PROTECTED) @Getter(AccessLevel.PROTECTED) FunctionName functionName;
     private @Setter(AccessLevel.PROTECTED) @Getter(AccessLevel.PROTECTED) Predictable predictor;
+    private @Getter(AccessLevel.PROTECTED) Set<String> targetWorkerNodes;
     private final Set<String> workerNodes;
     private final Queue<Double> modelInferenceDurationQueue;
     private final Queue<Double> predictRequestDurationQueue;
 
     public MLModelCache() {
+        targetWorkerNodes = ConcurrentHashMap.newKeySet();
         workerNodes = ConcurrentHashMap.newKeySet();
         modelInferenceDurationQueue = new ConcurrentLinkedQueue<>();
         predictRequestDurationQueue = new ConcurrentLinkedQueue<>();
     }
 
+    public void setTargetWorkerNodes(List<String> targetWorkerNodes) {
```

Comment:
Here targetWorkerNodes is a private concurrent set, so I assume MLModelCache needs to be threadsafe. In this case, if setTargetWorkerNodes can be run by multi-threads, we need to protect it by synchronize, otherwise, the targetWorkerNodes could be wrong.

Reply:
This method will be used by `MLModelCacheHelper.initModelState` method, which already has `synchronize`